azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/repository/RepositoryMiddleware.kt

```diff
@@ -35,13 +36,26 @@ internal class RepositoryMiddlewareImpl(
                 is ChatAction.MessageReceived -> processNewMessage(action, store::dispatch)
                 is ChatAction.MessageDeleted -> processDeletedMessage(action, store::dispatch)
                 is ChatAction.MessageEdited -> processEditMessage(action, store::dispatch)
+                is NetworkAction.Disconnected -> processNetworkDisconnected(store::dispatch)
             }
 
             // Pass Action down the chain
             next(action)
         }
     }
 
+    private fun processNetworkDisconnected(
```

Comment:
Not sure this belongs in this class, as it's not instrumenting MessageRepository. (this should just be installing/removing/editing/deleting messages in the repo)

Maybe we need NetworkMiddleware?

Reply:
resolving as per discussion